internal/ldso-cache/ldsocache.go

```diff
@@ -129,6 +129,103 @@ type LDSOCacheFile struct {
 	Extensions []LDSOCacheExtensionSection
 }
 
+type elfInfo struct {
+	Machine elf.Machine `yaml:"machine"`
+	Sonames []string    `yaml:"sonames"`
+}
+
+func doGetElfInfo(libfReaderAt io.ReaderAt) (elfInfo, error) {
+	info := elfInfo{}
+
+	elflibf, err := elf.NewFile(libfReaderAt)
+	if err != nil {
+		return info, err
+	}
+	// FIXME: do we need to check for the ELF magic bytes?
```

Comment:
Yes, it is security vulnerability otherwise. see how fun files were used to do remote code execution via unsuccessful ssh connections.

Reply:
thanks for reminding me of this. I agree that this should be checked at some level, and I wondered if debug/elf already guaranteed this. I think we're good:
https://cs.opensource.google/go/go/+/refs/tags/go1.24.3:src/debug/elf/file.go;drc=95b433eed428afbb4ab32f0f2541774e939989c7;l=294